components/state.js

```diff
@@ -0,0 +1,26 @@
+// state.js
+
+var _ = require('underscore');
+
+var Lm = require('./Lm');
+
+var states = [];
+
+// Returns the vni associated with the specified vnid.
+module.exports = function( inportName, sourceNodeName, sourcePortName, newState ) { 
```

Comment:
I'll have to see it.  Since this is a stub, I may not understand your intent.  If this were the real module for managing state objects, then I think it should only deal with state objects themselves -- not the inputStates object.  Parameters like sourceNodeName and sourcePortName only apply to the inputStates object.  I think the inputStates object should either be managed in the VNI module or in its own module, since a state object does not need to know about the inputStates object.